### ISSUE TITLE
No sprod_Denoised_matrix.txt and other output file but sprod.log under pseudo image option

### ISSUE THREAD

**Issue by yangj99** [yunguan-wang/SPROD#40]
Hi,

Thank you for this great software.
I have encountered one problem when running my sample data with no image.tif:
I got codes all executed in the terminal in the preset environment, but only acquired a UMAP_clustering.pdf, pseudo_image_features.csv in the input file with sprod.log in the output file. I am confused if there is any error that resulted in the absence of sprod_Denoised_matrix.txt, sprod_Detected_graph.txt, and sprod_Latent_space.txt.
Here is my sprod.log looks like:
`14:07:53,INFO:::Cold start, processing from counts and image data.`
`14:07:53,INFO:::Use spot cluster probability as pseudo image features`
`14:07:53,INFO:::Loading counts data and performing dimension reduction using UMAP.`
`14:07:55,INFO:::tick1 ok`
`14:07:55,INFO:::cts.shape[1] is 968`
`14:07:55,INFO:::hvg are Index(['CALR', 'EFHD1', 'KDM1A', 'SFRP1', 'ADGRL4', 'IGLC2', 'HNRNPA3',`
`14:07:55,INFO:::       'PPP1CA', 'RPL14', 'BST2',`
`14:07:55,INFO:::       ...`
`14:07:55,INFO:::       'SEC61B', 'ENY2', 'PTK7', 'HSP90AA1', 'SET', 'DHX9', 'RPS3', 'ANGPTL4',`
`14:07:55,INFO:::       'HLA-DRB1', 'EPCAM'],`
`14:07:55,INFO:::      dtype='object', length=968)`
`14:08:26,INFO:::tick 2 finishing UMAP dim reduction`
`14:09:58,INFO:::tick1`
`14:09:58,INFO:::2. Sprod finishing cold start check`
`14:09:58,INFO:::3. Sprod start denoising`
`14:09:58,INFO:::Proceed to Sprod denoising.`
`14:09:58,INFO:::Sprod-ready data format is single.`
`14:09:58,INFO:::Image derived features not found, will use pseudo image features.`
`14:09:58,INFO:::Total number of sprod jobs : 1`
`14:09:58,INFO:::['Rscript', 'D:\\JIAOYANG\\DESKTOPS\\SPROD\\sprod\\denoise.R', '-e',
'D:\\JIAOYANG\\DESKTOPS\\SPROD\\jyrep\\Dataset20\\input\\Counts.txt', '-c', 'D:\\JIAOYANG\\DESKTOPS\\SPROD\\jyrep\\Dataset20\\input\\Spot_metadata.csv', '-f', 'D:\\JIAOYANG\\DESKTOPS\\SPROD\\jyrep\\Dataset20\\input\\pseudo_image_features.csv', '-n', '-1', '-r', '0.08', '-u', '250', '-k', '10', '-l', '1', '-t', '0.4', '-m', '0.001', '-s', 'D:\\JIAOYANG\\DESKTOPS\\SPROD\\sprod', '-o', 'D:\\JIAOYANG\\DESKTOPS\\SPROD\\jyrep\\Dataset20\\output', '-p', 'sprod']`
`14:10:46,INFO:::4. Sprod start last step: stiching subsamples`
`14:10:46,INFO:::5. Sprod executing codes costs 172.76308512687683 seconds`

Basically, I just add some step marks to make sure each step is going on correctly without changing the codes too much.
The command feedback is literally just iterations.
I cannnot see where the problem occurred.  If you need my sample data to see, please let me know.

**Comment by yangj99** [comment]
Also, when I go through codes, I found the parameter "fr" called in the denoise.R file is confusing as I cannot refer it to anywhere in the script.  I am not sure if this has resulted in the absence of the output file. But as I tested, the Sprod.log stopped around this point. 
<img width="1046" alt="image" src="https://github.com/yunguan-wang/SPROD/assets/65137824/7cb855e8-8dea-4024-acd2-c7a6f3ed8198">
